lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -663,8 +663,25 @@ sub done {
     $validation->optional('result');
     $validation->optional('reason');
     $validation->optional('newbuild')->num(1);
+    $validation->optional('worker_id')->num(0, undef);
     return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
 
+    # check whether the specified worker matches the actually assigned one; refuse the update if not
+    if (my $specified_worker_id = $validation->param('worker_id')) {
+        my $assigned_worker_id = $job->assigned_worker_id;
+        if (defined $assigned_worker_id) {
+            return $self->render(
+                status => 400,
+                json => {error => "Refusing to set result because job is now assigned to worker $assigned_worker_id)."}
```

Comment:
```suggestion
                json => {error => "Refusing to set result because job is already assigned to worker $assigned_worker_id)."}
```

The error reflects a pre-existing condition, doesn't? Under that assumption I would find *now* to mean something else.

Reply:
This is basically the web UI saying to the worker that the job is *now* assigned to a different worker. The code does not do any checks about the past so I naturally used "now" in the error message. Besides, the job has very likely been assigned to this worker in the past but now that's no longer the case.